Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -141,6 +142,7 @@ public IonSerializer() : this(new IonSerializationOptions())
         public IonSerializer(IonSerializationOptions options)
         {
             this.options = options;
+            this.currentDepth = 0;

```

Comment:
This is not thread safe.

Reply:
currentDepth is no longer tracked by IonSerializer.